java-checks/src/main/java/org/sonar/java/checks/KnownCapacityHashBasedCollectionCheck.java

```diff
@@ -47,7 +50,12 @@ public List<Kind> nodesToVisit() {
 
   @Override
   protected void onConstructorFound(NewClassTree newClassTree) {
-    reportIssue(newClassTree, getIssueMessage(newClassTree));
+    QuickFixHelper.newIssue(context)
+      .forRule(this)
+      .onTree(newClassTree)
+      .withMessage(getIssueMessage(newClassTree))
+      .withQuickFixes(() -> computeQuickFix(newClassTree))
```

Comment:
Here we can use the singular api `withQuickFix` since we always provide only one